### PR TITLE
chore(repositories): Add yaml with all repositories used in Eclipse Che

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Introduce a file with list of components repositories used by Eclipse Che
This list was previously inside CONTRIBUTING.md file. Extract it to yaml file and allow update of CONTRIBUTING.md with a script

This PR includes
- yaml file with repositories list
- script to update (or check) CONTRINBUTING.md
- sort components by their names in CONTRIBUTING.md
- github action to check if CONTRIBUTING.md need to be updated or not (triggered only if the list of repositories is updated)

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17421

Change-Id: Idf551383bb806e11308d1e157af2f2921c593cd3
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
not tested but looks good conceptually. 

This will also provide us a single file which we can use to begin the process of transitioning from "master" to "main" and "wsmaster" to "wsserver", etc.

**Comment by nickboldt** [comment]
[crw-ci-test]

**Comment by ericwill** [comment]
Should we include the sidecar repos too?

**Comment by benoitf** [comment]
@ericwill the main issue with sidecar repositories is that they're not tied to a che version.
it'll be when you'll merge a PR in plugin-registry/devfile registry that you'll add reference it

**Comment by ericwill** [comment]
> @ericwill the main issue with sidecar repositories is that they're not tied to a che version.
> it'll be when you'll merge a PR in plugin-registry/devfile registry that you'll add a reference to it

Will the PR's for the sidecar repos then be included in a list for PR's of some milestone? I ask because usually the sidecar update work is what takes longer, whereas updating the `meta.yaml` doesn't take as long. It's therefore important the sidecar PR's get reported somewhere.

File: .github/workflows/contributing-repositories-check.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - '.repositories.yaml'
+  pull_request:
+    paths:
+      - '.repositories.yaml'
+jobs:
+  check-contributing-up-to-date:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout che-theia source code
```
Comment (l0rd) [inline .github/workflows/contributing-repositories-check.yml:26]: che-theia?

File: .repositories.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    description: >-
+      (this repository) the main project repository
+    devfile: https://github.com/eclipse/che/blob/master/devfile.yaml
+    doc: https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#che-server-aka-ws-master
+    components: 
+     - name: dockerfiles
+       url: https://github.com/eclipse/che/tree/master/dockerfiles
+       description: >-
+         source code, dockerfiles to build our main docker images. Note that Che-theia related dockerfiles are located in che-theia repo.
+     - name: orchestrator
```
Comment (l0rd) [inline .repositories.yaml]: controller?

Comment (l0rd) [reply .repositories.yaml]: or che-api

**Review by l0rd — APPROVED** [review]
This is really good @benoitf. Something that we discussed but I cannot see is an attribute that would specify if a reposiotory follow eclipse/che release lifecycle or not (i.e. if we should create milestones or at every release or not).

**Comment by ericwill** [comment]
> > @ericwill the main issue with sidecar repositories is that they're not tied to a che version.
> > it'll be when you'll merge a PR in plugin-registry/devfile registry that you'll add a reference to it
> 
> I ask because usually the sidecar update work is what takes longer, whereas updating the `meta.yaml` doesn't take as long. It's therefore important the sidecar PR's get reported somewhere.

Better said, the sidecar updates are usually where newsworthy things land, for example "OpenShift connector plugin now uses OpenShift 4" is a sidecar PR, not plugin registry.

**Comment by benoitf** [comment]
Eric, Mario
 I will add all repositories but introduce a new flag like `cheMilestone` `true|false` to know if it's part or not of same release lifecycle

If you have better name for this flag, please raise your hand :-)

**Comment by ericwill** [comment]
> Eric, Mario
> I will add all repositories but introduce a new flag like `cheMilestone` `true|false` to know if it's part or not of same release lifecycle
> 
> If you have better name for this flag, please raise your hand :-)

Sounds good :+1:

File: .repositories.yaml (hunk 2 above)
Comment (benoitf) [reply .repositories.yaml]: I keep for now wsmaster, will track change of the name in another issue

**Comment by benoitf** [comment]
Updated with more repositories

introduced `useCheReleaseLifecycle` attribute

**Comment by benoitf** [comment]
@ericwill I've added repositories from che-dockerfiles

@l0rd ok for `useCheReleaseLifecycle` or you've another proposal ?

**Pull request merged**